src/lib/components/popup.component.ts

```diff
@@ -0,0 +1,97 @@
+import { defineComponent, inject, onMounted, onBeforeUnmount, PropType, unref, watch, ref, h } from 'vue';
+import { LngLatLike, Popup, Offset, PositionAnchor, PopupOptions, PointLike } from 'maplibre-gl';
+import { MapLib } from '@/lib/lib/map.lib';
+import { mapSymbol, markerSymbol } from '@/lib/types';
+
+export default /*#__PURE__*/ defineComponent({
+  name : 'MglPopup',
+  emits: ['open', 'close'],
+  props: {
+    coordinates: {
+      type: [ Object, Array ] as unknown as PropType<LngLatLike>,
+      required: false
+    },
+    closeButton: {
+      type: Boolean,
+      required: false,
+      default: true,
+    },
+    closeOnClick: {
+      type: Boolean,
+      required: false,
+      default: true,
+    },
+    closeOnMove: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    focusAfterOpen: {
+      type: Boolean,
+      required: false,
+      default: true,
+    },
+    anchor: {
+      type: String as PropType<PositionAnchor>,
+      required: false
+    },
+    offset: {
+      type: [Number, Object, Array] as PropType<Offset>,
+      required: false,
+    },
+    className: {
+      type: String,
+      required: false,
+    },
+    maxWidth: {
+      type: String,
+      default: '240px',
+    },
+    text: {
+      type: String,
+      required: false
+    }
+  },
+  async setup(props, { slots, emit }) {
```

Comment:
Is the async needed? There is no await keyword.

Reply:
This should be flagged with linting no?

Reply:
The async is not needed here. I fixed it on my fork: https://github.com/indoorequal/vue-maplibre-gl